### PR TITLE
clojure: Fix file specific variable

### PR DESCRIPTION
**Pull request by lebensterben**
- Labelled `clojure-backend` as safe local variable.
- Added local variable hooks of clojure mode:
  - `spacemacs//clojure-setup-backend`
- Added new function `spacemacs//clojure-bindings-hook` to setup
  prefixes and bindings. Common prefixes and bindings are set only
  once. `cider` specific prefixes and bindings are set after
  file local variables are loaded to allow local override of
  `clojure-backend`.
- `spacemacs|forall-clojure-modes` is renamed to
  `spacemacs||forall-clojure-modes` according to naming conventions.

See: https://github.com/syl20bnr/spacemacs/issues/14653

-------

Thank you :heart: for contributing to Spacemacs!

Before you submit this pull request, please ensure it is against the `develop` branch and that you have read the [CONTRIBUTING.org](https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org) file. It contains instructions on how to properly follow our conventions on commit messages, documentation, change log entries and other elements.

Don't forget to update CHANGELOG.develop if you want to be mentioned in the release file!

This message should be replaced with a description of your change.

Cheers!

### CODE REVIEW

**Comment by lebensterben** [comment]
Found some issue. Hold on.

File: layers/+lang/clojure/funcs.el [hunk 1]
```diff
@@ -224,7 +242,7 @@ in your Spacemacs configuration:
              clojure-enable-clj-refactor t)
     ) ")))
 
-(defmacro spacemacs|forall-clojure-modes (m &rest body)
+(defmacro spacemacs||forall-clojure-modes (m &rest body)
```
Comment (syl20bnr) [inline layers/+lang/clojure/funcs.el:245]: Maybe we can keep it public as users might find this macro handy for their custom config.

Comment (smile13241324) [reply layers/+lang/clojure/funcs.el:245]: Yep, btw we should add a line to the conventions defining that a macro with `||` is private. Right now we only have this for functions and variables.

File: layers/+lang/clojure/funcs.el [hunk 2]
```diff
@@ -25,6 +25,24 @@
   "Conditionally setup clojure backend."
   (when (eq clojure-backend 'lsp) (lsp)))
 
+(defun spacemacs//clojure-bindings-hook
+    (common-prefix cider-prefix common-binding cider-binding)
+  "Conditinally setup clojure-modes bindings."
```
Comment (smile13241324) [inline layers/+lang/clojure/funcs.el:30]: Small typo
```suggestion
  "Conditionally setup clojure-modes bindings."
```

**Review by smile13241324 — CHANGES_REQUESTED** [review]
Found a small typo, the rest looks good. Can you check @syl20bnr comment? Maybe we should keep this macro non private for private layers.

**Pull request closed without merging**